probe_src/probe_py/analysis.py

```diff
@@ -47,71 +49,77 @@ def first(pid: int, exid: int, tid: int) -> Op:
 
     def last(pid: int, exid: int, tid: int) -> Node:
         if not proc_to_ops.get((pid, exid, tid)):
-            exit_node = (pid, exid, tid, "<exit>")
+            # as Op object is not availaible, op_index will be -1
+            exit_node = (pid, exid, tid, "<exit>", -1)
             proc_to_ops[(pid, exid, tid)] = [exit_node]
             nodes.append(exit_node)
             return exit_node
         else:
             return proc_to_ops[(pid, exid, tid)][-1]
 
-    fork_join_edges = list[tuple[Node, Node]]()
-    exec_edges = list[tuple[Node, Node]]()
-
     # Hook up forks/joins
     for node in list(nodes):
-        pid, exid, tid, op = node
-        if False:
-            pass
-        elif isinstance(op, CloneOp):
+        pid, exid, tid, op, op_index = node
+        if isinstance(op, CloneOp):
             if op.flags & CLONE_THREAD:
                 # Spawning a thread links to the current PID and exec epoch
                 target = (pid, exid, op.child_thread_id)
             else:
-                # New process always links to exec epoch 0 and main thread
-                # THe TID of the main thread is the same as the PID
-                target = (op.child_process_id, 0, op.child_process_id)
+                target = (op.child_process_id, 0, 0)
             exec_edges.append((node, first(*target)))
         elif isinstance(op, WaitOp) and op.ferrno == 0 and op.ret > 0:
             # Always wait for main thread of the last exec epoch
             if op.ferrno == 0:
-                target = (op.ret, last_exec_epoch.get(op.ret, 0), op.ret)
+                target = (op.ret, -1, last_exec_epoch.get(op.ret, 0), op.ret)
                 fork_join_edges.append((last(*target), node))
         elif isinstance(op, ExecOp):
             # Exec brings same pid, incremented exid, and main thread
             target = pid, exid + 1, 0
```

Comment:
I noticed it while reviewing the code, here the `tid` should not be 0 as the main thread has TID == PID. Should I change it to `pid` ? @charmoniumQ

Reply:
Yes plz